### ISSUE TITLE
[Enhancement]: aws_msk_serverless_cluster to return bootstrap server url

### ISSUE THREAD

**Issue by Fran-Rg** [hashicorp/terraform-provider-aws#28005]
### Description

The aws_msk_serverless_cluster recently added doesn't return the URL of the created cluster.

Now this ticket will need to be on hold, pending on aws to provide a way to programmatically retrieve that information.

As far as I can see it's not available: [describe-cluster-v2](https://docs.aws.amazon.com/cli/latest/reference/kafka/describe-cluster-v2.html) returned data doesn't have the information.

And there are no method related to client-information:
https://docs.aws.amazon.com/cli/latest/reference/kafka/index.html

### Affected Resource(s) and/or Data Source(s)

* aws_msk_serverless_cluster

### Potential Terraform Configuration

```terraform
* aws_msk_serverless_cluster.client_information
```

### References

https://github.com/hashicorp/terraform-provider-aws/issues/22058

### Would you like to implement a fix?

No

**Comment by Fran-Rg** [comment]
Raised it with aws, it's possible to get that information via:
```
aws kafka  get-bootstrap-brokers --cluster-arn "<CLUSTER_ARN>"
{
    "BootstrapBrokerStringSaslIam": "<URL>:9098"
}
```

https://docs.aws.amazon.com/msk/1.0/apireference/clusters-clusterarn-bootstrap-brokers.html

Either this get fetched as post resource creation or a new data is created

**Comment by yermulnik** [comment]
We've just hit this (inability to programmatically derive Serverless bootstrap endpoint) 😞
Looking forward for this to be implemented soon 🤞🏻

**Comment by yermulnik** [comment]
Apart from amending `aws_msk_serverless_cluster` resource by adding cluster connection info, it also would be very much useful either to have `msk_broker_nodes` data source amended respectively or to have a dedicated data source for Serverless MSK. Thank you.

**Comment by rpadaki** [comment]
So would this be as simple as adding the following to `resourceServerlessClusterRead` in serverless_cluster.go?

```go
	brokers, _ := conn.GetBootstrapBrokers(&kafka.GetBootstrapBrokersInput{
		ClusterArn: cluster.ClusterArn,
	})
	
	d.Set("bootstrap_brokers", brokers.BootstrapBrokerString)
	d.Set("bootstrap_brokers_sasl_iam", brokers.BootstrapBrokerStringSaslIam)
	d.Set("bootstrap_brokers_public_tls", brokers.BootstrapBrokerStringPublicTls)
	...
```

If that's all it takes, I'm happy to put up a PR sometime in the next couple of weeks

**Comment by kaykhan** [comment]
I started a PR to create a new datasource that will get the endpoint via `get-bootstrap-brokers` request - https://docs.aws.amazon.com/cli/latest/reference/kafka/get-bootstrap-brokers.html


My first PR, so i will need help. please upvote to help with prioritization!

https://github.com/hashicorp/terraform-provider-aws/pull/32484

**Comment by yermulnik** [comment]
@ewbankkit The #32484 seems to only partially resolve this request which initially was to implement this feature via **resource** rather than a *data source** (while having data source that also does that is also desired). Can you please re-open this issue so that maybe someone can contribute by implementing Serverless MSK brokers list within `aws_msk_serverless_cluster` resource? Thanks.

@kaykhan I'm not entirely sure and hence am asking for your assistance: does `aws_msk_bootstrap_brokers` (implemented in #32484) duplicate already existing `aws_msk_broker_nodes` with an enhanced feature to also provide broker nodes info for Serverless MSK along with Provisioned MSK or does it provide broker nodes info **only** for Serverless MSK? Thanks.
It also probably makes sense to add cross-references to the documentation of `aws_msk_bootstrap_brokers` and `aws_msk_broker_nodes` data sources to eliminate confusion of why two of the sane kind exist. Thanks.

**Comment by yermulnik** [comment]
@breathingdust Could you please look into re-opening this issue since it was resolved partially and sort of indirectly? Thanks.
